core/http/src/main/java/org/trellisldp/http/impl/MementoResource.java

```diff
@@ -107,7 +107,7 @@ public ResponseBuilder getTimeMapBuilder(final SortedSet<Instant> mementos, fina
             final String baseUrl) {
 
         final List<MediaType> acceptableTypes = req.getAcceptableMediaTypes();
-        final String identifier = getBaseUrl(baseUrl, req) + req.getPath();
+        final String identifier = fromUri(getBaseUrl(baseUrl, req)).path(req.getPath()).build().toString();
```

Comment:
I haven't spent much time at all in the Trellis code, but I wonder whether making this change without a related change in the test suite (to prevent regressions) might come back to bite us. 

Thanks for addressing these issues so quickly, @acoburn!

Reply:
Yes, adding some regression tests for this is a very good idea :-)

Reply:
Additional tests are on the way -- with the new tests, the timemap tests failed (expectedly) when run against the current `master` branch, but they succeed with this modification in place.